packet/member.py

```diff
@@ -1,11 +1,188 @@
-from .models import Freshman, FreshSignature, UpperSignature, MiscSignature
+from collections import namedtuple
+from logging import getLogger
 
+from sqlalchemy import exc
 
-def signed_packets(member):
-    # Checks whether or not member is a freshman
-    if Freshman.query.filter_by(rit_username=member).first() is not None:
-        return FreshSignature.query.filter_by(freshman_username=member, signed=True).all()
-    # Checks whether or not member is an upperclassman
-    if UpperSignature.query.filter_by(member=member).first() is not None:
-        return UpperSignature.query.filter_by(member=member, signed=True).all()
-    return MiscSignature.query.filter_by(member=member).all()
+from .models import db, REQUIRED_MISC_SIGNATURES
+from .packet import get_number_required, get_misc_signatures
+
+LOGGER = getLogger(__name__)
+
+
+def current_packets(member, intro=False, onfloor=False):
+    """
+    Get a list of currently open packets with the signed state of each packet.  Returns a <result>
+    :param member: the member currently viewing all packets
+    :param intro: true if current member is an intro member
+    :param other: true if current member is off floor or alumni
+    :return: <tuple> a list of packets that are currently open, and their attributes
+    """
+
+    # Tuple for compatibility with UI code.  Should be refactored or deleted altogether later
+    SPacket = namedtuple('spacket', ['rit_username', 'name', 'did_sign', 'total_signatures', 'required_signatures'])
```

Comment:
I haven't looked at devin's refactor on #69 so i'm going to just leave it for now.  After both have been merged in I can look at it because it requires changing the UI code for the templates